pkg/segment/structs/segsearchstructs.go

```diff
@@ -375,6 +375,26 @@ func extractSearchQueryFromMatchFilter(match *MatchFilter) *SearchQuery {
 	} else {
 		currQuery.SearchType = MatchWords
 	}
+	if len(match.MatchWords) > 0 {
+		for _, word := range match.MatchWords {
+			cval := dtu.ReplaceWildcardStarWithRegex(string(word))
+			rexpC, err := regexp.Compile(cval)
+			if err != nil {
+				log.Errorf("extractSearchQueryFromMatchFilter: regexp compile failed for MatchWord %s, err=%v", word, err)
+			} else {
+				currQuery.MatchFilter.Regexp = rexpC
+			}
+		}
+	} else if match.MatchPhrase != nil {
+		cval := dtu.ReplaceWildcardStarWithRegex(string(match.MatchPhrase))
```

Comment:
apart from checking for `match.MatchPhrase` not being nill, you should also check if `match.MatchPhrase` contains a `*`, only then fill in the regexp , that way we don't unnecessarily to do regex search for exact match